test/e2e/resize_nodes.go

```diff
@@ -469,68 +460,67 @@ var _ = Describe("Nodes", func() {
 	})
 
 	Describe("Network", func() {
-		BeforeEach(func() {
-			if !providerIs(supportedProviders...) {
-				Failf("Nodes.Network test is only supported for providers %v (not %s). You can avoid this failure by using ginkgo.skip=Nodes.Network in your environment.",
-					supportedProviders, testContext.Provider)
-			}
-		})
-
-		// TODO marekbiskup 2015-06-19 #10085
-		// This test has nothing to do with resizing nodes so it should be moved elsewhere.
-		// Two things are tested here:
-		// 1. pods from a uncontactable nodes are rescheduled
-		// 2. when a node joins the cluster, it can host new pods.
-		// Factor out the cases into two separate tests.
-
-		testName = "Uncontactable nodes, have their pods recreated by a replication controller, and can host new pods after rejoining."
-		It(testName, func() {
-			if testContext.CloudConfig.NumNodes < 2 {
-				By(fmt.Sprintf("skipping %s test, which requires at least 2 nodes (not %d)",
-					testName, testContext.CloudConfig.NumNodes))
-				return
-			}
-
-			// Create a replication controller for a service that serves its hostname.
-			// The source for the Docker containter kubernetes/serve_hostname is in contrib/for-demos/serve_hostname
-			name := "my-hostname-net"
-			newSVCByName(c, ns, name)
-			replicas := testContext.CloudConfig.NumNodes
-			newRCByName(c, ns, name, replicas)
-			err := verifyPods(c, ns, name, true, replicas)
-			Expect(err).NotTo(HaveOccurred(), "Each pod should start running and responding")
-
-			By("choose a node with at least one pod - we will block some network traffic on this node")
-			label := labels.SelectorFromSet(labels.Set(map[string]string{"name": name}))
-			pods, err := c.Pods(ns).List(label, fields.Everything()) // list pods after all have been scheduled
-			Expect(err).NotTo(HaveOccurred())
-			nodeName := pods.Items[0].Spec.NodeName
-
-			node, err := c.Nodes().Get(nodeName)
-			Expect(err).NotTo(HaveOccurred())
+		Context("when a minion node becomes unreachable", func() {
+			BeforeEach(func() {
+				supportedProviders := []string{"gce", "gke"}
+				if !providerIs(supportedProviders...) {
+					Skipf("Only supported for providers %v (not %s)", supportedProviders, testContext.Provider)
+				}
+				if testContext.CloudConfig.NumNodes < 2 {
+					Skipf("Requires at lease 2 nodes (not %d)", testContext.CloudConfig.NumNodes)
```

Comment:
least.

it seems like the node number check is also a common pattern that may be worth turning into a utility function.